framework/encode/capture_manager.h

```diff
@@ -148,6 +141,8 @@ class CaptureManager
     bool GetIUnknownWrappingSetting() const { return iunknown_wrapping_; }
 
   protected:
+    typedef std::shared_mutex ApiCallMutexT;
+
```

Comment:
Typedef the lock type too?
```c++
typedef std::shared_lock<util::SharedMutex> ApiCallLockT;
```
Just a thought. The uses of the lock are always `auto` anyway.

Reply:
Each lock type (`std::shared_lock<ApiCallMutexT>`, `std::unique_lock<ApiCallMutexT>`) is defined just once in each of the `Acquire*ApiCallLock` methods so I think their typedefs can be omitted.